aten/src/ATen/native/Fill.cpp

```diff
@@ -121,19 +120,19 @@ Tensor& fill_diagonal_(Tensor& self, const Scalar& fill_value, bool wrap) {
   strides.push_back(stride);
   sizes.push_back(size);
 
-  auto main_diag = self.as_strided(sizes, strides, storage_offset);
+  auto main_diag = self.as_strided_symint(sizes, strides, storage_offset);
   main_diag.fill_(fill_value);
 
   if (wrap && nDims == 2 && height > width + 1) {
-    std::vector<int64_t> wrap_sizes;
+    std::vector<SymInt> wrap_sizes;
 
-    int64_t step = width + 1;
-    int64_t wrap_size = ((self.numel() + step - 1) / step) - size;
+    auto step = width + 1;
+    auto wrap_size = ((self.numel() + step - 1) / step) - size;
     wrap_sizes.push_back(wrap_size);
```

Comment:
Nit should be emplace_back or better yet just construct using a list initialize since it's a single element vector that's not referenced earlier, just construct the vector here with brace syntax.